crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -584,17 +584,19 @@ impl<F: Clone> PostUpdateTracker<F, PaymentData<F>, types::PaymentsSyncData> for
                         (
                             mandate_ref.connector_mandate_id.clone(),
                             mandate_ref.mandate_metadata.clone(),
+                            mandate_ref.connector_mandate_request_reference_id.clone(),
```

Comment:
There would be a case where we do not get connector_mandate_request_reference_id back in the response from the connector, then the value which we've set for connector_mandate_request_reference_id would get overridden right? That's why we should always take source and populate the original value if the current value is None @awasthi21

Reply:
Added the check